### PR TITLE
Initial Kafka channel reconciler tests

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Pushing for early feedback
- Check individual commits
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/channel/channel_test.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	DataPlaneConfigMapName:      "kafka-channel-channels-subscriptions",
+	GeneralConfigMapName:        "kafka-broker-config",
+	IngressName:                 "kafka-channel-ingress",
+	SystemNamespace:             "knative-eventing",
+	DataPlaneConfigFormat:       base.Json,
+}
+
+// TODO: tests with and without subscriptions
+// TODO: tests for things from config-kafka (partition count, etc)
+// TODO: are we setting a InitialOffsetsCommitted status?
```
Comment (aliok) [inline control-plane/pkg/reconciler/channel/channel_test.go:119]: This PR contains a single test, will add more later

File: control-plane/pkg/reconciler/channel/channel_test.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+							BootstrapServers: ChannelBootstrapServers,
+							Ingress: &contract.Ingress{
+								IngressType: &contract.Ingress_Path{
+									Path: receiver.Path(ChannelNamespace, ChannelName),
+								},
+							},
+							// TODO: do we need to set this?
+							//Auth:      &contract.Resource_AbsentAuth{},
+							// TODO: do we need to set this?
+							//Reference: &contract.Reference{Namespace: ChannelNamespace, Name: ChannelName},
```
Comment (aliok) [inline control-plane/pkg/reconciler/channel/channel_test.go:161]: Do we need to set these for the channel? I see they're not set in broker+sink but set in source

File: control-plane/pkg/reconciler/channel/channel_test.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+						NumPartitions:     1,
+						ReplicationFactor: 1,
+					},
+					// TODO: how much of these we need and want here?
+					ErrorOnCreateTopic:            nil,
+					ErrorOnDeleteTopic:            nil,
+					ExpectedClose:                 false,
+					ExpectedCloseError:            nil,
+					ExpectedErrorOnDescribeTopics: nil,
+					//ExpectedTopicsMetadataOnDescribeTopics: topicMetadata,
```
Comment (aliok) [inline control-plane/pkg/reconciler/channel/channel_test.go:240]: I am guessing we wouldn't need this.

File: control-plane/pkg/reconciler/channel/channel_test.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/channel/channel_test.go:161]: No, there is no need for them.

File: control-plane/pkg/reconciler/channel/channel_test.go (hunk 3 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/channel/channel_test.go:240]: Right, we don't need this.

**Comment by aliok** [comment]
@pierDipi 
I removed the WIP status. I want to merge this PR and add more tests later (I will NOT do a separate PR for each test case, don't worry 😄  ).

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**